src/Tracking/Events/GivePluginSettingsTracking.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace Give\Tracking\Events;
+
+use Give\Tracking\Contracts\TrackEvent;
+use Give\Tracking\Track;
+use Give\Tracking\TrackingData\GivePluginSettingsData;
+
+/**
+ * Class GivePluginSettingsTracking
+ *
+ * This class setup event to send tracked data request when Give plugin settings update.
+ *
+ * @since 2.10.0
+ * @package Give\Tracking\Events
+ */
+class GivePluginSettingsTracking implements TrackEvent {
+	/**
+	 * @inheritdoc
+	 */
+	public function boot() {
+		add_action( 'update_option_give_settings', [ $this, 'record' ] );
+	}
+
+	/**
+	 * Send track
+	 *
+	 * @since 2.10.0
+	 */
+	public function record() {
+		/* @var Track $track */
+		$track = give( Track::class );
```

Comment:
@kjohnson I am doing that because I am registering for this class as singlton and this class store/keep event which we send to the server on `shutdown` hook. Instead of making this as a singleton, I can make the static property to keep a record of all events. Do you have any suggestions?

Reply:
A singleton will still work with dependency injection. The class will be instantiated and then the container will return the same instance, so the injected `Track::class` will still be available.

Reply:
So long as a class is instantiated using the Service Container, you can include dependencies in the constructor and they'll be provided to the instance when it's created.

Reply:
@kjohnson I refactor code as per your suggestion: https://github.com/impress-org/givewp/compare/2c29e031b44d...811ca90b6586